### PR TITLE
CP-17230: Create UI design for proxy authentication

### PR DESCRIPTION
**Pull request by Frezzle**
Added 2 textboxes for proxy username and password; encrypted when saved
and decrypted when loaded. Not logged.
Added 2 checkboxes; one for bypassing proxy for xenserver connections, one
for providing proxy credentials.
Controls are no longer enabled/disabled when the related proxy option is
selected/deselected.
Making changes to a control (e.g. username, bypass checkbox, etc) selects
the 'Use this proxy server' option automatically.

### CODE REVIEW

File: XenAdmin/Dialogs/OptionsPages/ConnectionOptionsPage.cs [hunk 1]
```diff
@@ -170,11 +228,31 @@ public void Save()
             if (ProxyAddressTextBox.Text != Properties.Settings.Default.ProxyAddress && ProxyAddressTextBox.Text != "")
                 Properties.Settings.Default.ProxyAddress = ProxyAddressTextBox.Text;
 
-            if (new_proxy_style == HTTPHelper.ProxyStyle.SpecifiedProxy)
+            Properties.Settings.Default.ProxyUsername = EncryptionUtils.Protect(ProxyUsernameTextBox.Text);
+            Properties.Settings.Default.ProxyPassword = EncryptionUtils.Protect(ProxyPasswordTextBox.Text);
+
+            if (int.Parse(ProxyPortTextBox.Text) != Properties.Settings.Default.ProxyPort)
```
Comment (kc284) [inline XenAdmin/Dialogs/OptionsPages/ConnectionOptionsPage.cs]: No exception handling in line 234. Also, consider refactoring to avoid parsing twice.

**Comment by kc284** [comment]
Just to be on the safe side, I would use IsStringNullOrEmpty in lines 96, 98, 175, 228 instead of checking only for empty.

**Comment by Frezzle** [comment]
Thanks @kc284, I've made the changes.

**Comment by kc284** [comment]
@Frezzle looks good now.

**Comment by MihaelaStoica** [comment]
@Frezzle: Shall we close this PR, as it's superseded by #1031?

**Comment by Frezzle** [comment]
Let's wait for #1031 to be merged first? If not then it's fine to close I think.

**Pull request closed without merging**